libraries/abstractions/cellular/test/test_cellular_api.c

```diff
@@ -433,6 +437,13 @@ static BaseType_t prvConnectCellular( void )
     uint8_t NumStatus = 1;
     bool simReady = false;
 
+    /* Clean up the cellular handle before init. */
+    if( _cellularHandle != NULL )
+    {
+        ( void ) Cellular_Cleanup( _cellularHandle );
+        _cellularHandle = NULL;
```

Comment:
Do we need to set _cellularHandle  = NULL?
Can Cellular_Cleanup do _cellularHandle  = NULL? If not, maybe we need to update the description of the API?
"This function frees resources taken in Cellular_Init", _cellularHandle  is also allocated from Cellular_Init.

/**
 * @brief One time deinitialization function.
 *
 * This function frees resources taken in Cellular_Init. After this function
 * returns, Cellular_Init must be called again before calling any other function
 * of this library.
 *
 * @param[in] cellularHandle The opaque cellular context pointer created by Cellular_Init.
 *
 * @return CELLULAR_SUCCESS if the operation is successful, otherwise an error
 * code indicating the cause of the error.
 */
CellularError_t Cellular_Cleanup( CellularHandle_t cellularHandle );

Reply:
It is required to set _cellularHandle = NULL
The context allocated in Cellular_Init is freed in Cellular_Cleanup.
_cellularHandle is an opaque pointer.

On the other hand,
If we want Cellular_Cleanup clear the pointer value, we need to change the proto type of the Cellular_Cleanup to.
CellularError_t Cellular_Cleanup( CellularHandle_t cellularHandle );
=>
CellularError_t Cellular_Init( CellularHandle_t * pCellularHandle );